code/modules/mob/living/silicon/robot/robot_subtypes/starman.dm

```diff
@@ -215,4 +297,41 @@
 
 	user.visible_message("<span class='danger'>\The [user] bends reality in impossible ways!</span>","<span class='notice'>*Beep* Hostile consciousnesses twisted.</span>")
 
-	..()
\ No newline at end of file
+	..()
+	
+	
+/obj/item/weapon/gun/energy/starman_beam
+	name = "Psi Beam Vestibule"
+	desc = "Used to channel psionic energy into a deadly form."
+	icon_state = "pulse"
+	item_state = "gun"
+	fire_sound = 'sound/effects/psi/psi_beam.ogg'
+	fire_delay = 3 SECONDS
+	cell_type = "/obj/item/weapon/cell/infinite"
+	projectile_type = "/obj/item/projectile/beam/pulse"
```

Comment:
jesus christ

Reply:
Again, a reliable way to shoot shit is needed when you're an immobile rock. It's a medium-high danger adminbus, so keep that in mind.